### PR TITLE
fix kraken crash due to MON_IP with wrong value

### PR DESCRIPTION
**Pull request by shiywang**
Fixes: https://github.com/kubevirt/kubevirt-ansible/issues/232

@aglitke @qwang1 @gouyang PTAL, not sure if this is a corret way to fix it, but the root cause I guess is already as @qwang1 described

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by shiywang** [comment]
ci test please

**Comment by shiywang** [comment]
ci test please

**Comment by qwang1** [comment]
Because my previous testing-cluster enabled cloud-provider，the hostname，instance name，internal_ip are consistent, so I didn't encounter the problem.

**Comment by lukas-bednar** [comment]
ci add to whitelist

**Comment by shiywang** [comment]
ci test please

File: roles/storage-demo/templates/storage-demo.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ spec:
         args: ["demo"]
         env:
         - name: MON_IP
-          value: "{{ storage_demo_node_hostname }}"
+          value: "{{ storage_demo_node_ip }}"
```
Comment (aglitke) [inline roles/storage-demo/templates/storage-demo.yml]: Don't forget to fix it everywhere in the manifest

Comment (aglitke) [reply roles/storage-demo/templates/storage-demo.yml]: Also, did you try:
```
      - name: kraken                                                            
        args: ["demo"]                                                          
        env:                                                                    
        - name: MON_IP                                                          
          valueFrom:                                                            
            fieldRef:                                                           
              fieldPath: status.hostIP                                          
```

Comment (shiywang) [reply roles/storage-demo/templates/storage-demo.yml]: yeah, this format is same as ceph-cinder-demo, but it's seems to have some validation problem in cli or apiserver, I'll try to figure it out.

**Comment by shiywang** [comment]
@aglitke PTAL

**Comment by shiywang** [comment]
@gbenhaim PTAL

**Comment by shiywang** [comment]
@aglitke @screeley44 also mind take a look this one, I think it's ready for merge

**Comment by shiywang** [comment]
ci test please

**Comment by shiywang** [comment]
@aglitke could you please take a look ?

**Comment by aglitke** [comment]
Thanks!  PR looks good but please squash the commits.  There are 6 commits in this PR for just this one simple change.

**Pull request merged**